nixos/modules/services/web-apps/gotosocial.nix

```diff
@@ -0,0 +1,236 @@
+        '';
+        type = with lib.types; listOf str;
+        default = [ ];
+        example = [ "--media-image-max-size" "2097152" ];
+      };
+
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Open the configured port in the firewall.
+          We highly reccommend using a reverse proxy instead.
+        '';
+      };
+
+      database = {
+        createLocally = lib.mkOption {
+          description = lib.mdDoc "Configure local PostgreSQL database server for GotoSocial.";
+          type = lib.types.bool;
+          default = true;
+        };
+
+        host = lib.mkOption {
+          type = lib.types.str;
+          default = "/run/postgresql";
+          example = "192.168.23.42";
+          description = lib.mdDoc "Database host address or unix socket.";
+        };
+
+        port = lib.mkOption {
+          type = lib.types.int;
+          default = 5432;
+          description = lib.mdDoc "Database host port.";
+        };
+
+        name = lib.mkOption {
+          type = lib.types.str;
+          default = "gotosocial";
+          description = lib.mdDoc "Database name.";
+        };
+
+        user = lib.mkOption {
+          type = lib.types.str;
+          default = "gotosocial";
+          description = lib.mdDoc "Database user.";
+        };
+
+        passwordFile = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = null;
+          example = "/run/keys/gotosocial-db-password";
+          description = lib.mdDoc ''
+            A file containing the password corresponding to
+            {option}`database.user`.
+          '';
+        };
+      };
+
+      settings = lib.mkOption {
+        type = settingsFormat.type;
+        description = lib.mdDoc ''
+          Contents of the GotoSocial YAML config.
+          The documentation may be found
+          [here](https://docs.gotosocial.org/en/latest/configuration/),
+          and an example may be found
+          [here](https://github.com/superseriousbusiness/gotosocial/blob/main/example/config.yaml).
+          _Note_: the `db-type` and related fields are set automatically.
+        '';
+        default = {};
+        example = {
+          application-name = "gotosocial";
+          protocol = "https";
+          host = "localhost";
+          bind-address = "::";
+          port = 8080;
+          storage-local-base-path = "/var/lib/gotosocial/storage";
+        };
+      };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = databaseActuallyCreateLocally -> (cfg.user == cfg.database.user);
+        message = ''For local automatic database provisioning (services.mastodon.database.createLocally == true) with peer authentication (services.mastodon.database.host == "/run/postgresql") to work services.mastodon.user and services.mastodon.database.user must be identical.'';
```

Comment:
The usage of `mastodon` here looks like a copy/paste error and should probably be `gotosocial` I guess?